it/repositories/application/GeneralApplicationMongoRepositorySpec.scala

```diff
@@ -469,6 +469,41 @@ class GeneralApplicationMongoRepositorySpec extends MongoRepositorySpec with UUI
     }
   }
 
+  "Update application route" should {
+    "return not found if the application does not match required statuses" in {
+      testDataRepo.createApplicationWithAllFields(UserId, AppId, "FastStream-2016",
+        additionalDoc = BSONDocument("applicationRoute" -> ApplicationRoute.Edip.toString)
+      ).futureValue
+
+      val result = repository.updateApplicationRoute(AppId, ApplicationRoute.SdipFaststream).failed.futureValue
+
+      result mustBe a[Exceptions.NotFoundException]
+    }
+
+    "update the Faststream application" in {
```

Comment:
When the "update the Faststream application" happens? Maybe it will be clearer if you say:

update the Faststream application when condition is true